nautobot_chatops/workers/nautobot.py

```diff
@@ -1045,6 +1051,343 @@ def get_circuit_providers(dispatcher, *args):
+    """Get a filtered list of jobs from Nautobot that the request user have view permissions for.
+
+    Args:
+        job_filters (str): Filter job results by literals in a comma-separated string.
+                           Available filters are: enabled, installed.
+    """
+    # Check for filters in user supplied input
+    job_filters_list = [item.strip() for item in job_filters.split(",")] if isinstance(job_filters, str) else ""
+    filters = ["enabled", "installed"]
+    if any(key in job_filters for key in filters):
+        filter_args = {key: True for key in filters if key in job_filters_list}
+        jobs = Job.objects.restrict(dispatcher.user, "view").filter(**filter_args)  # enabled=True, installed=True
+    else:
+        jobs = Job.objects.restrict(dispatcher.user, "view").all()
+
+    header = ["Name", "ID", "Enabled"]
+    rows = [
+        (
+            str(job.name),
+            str(job.id),
+            str(job.enabled),
+        )
+        for job in jobs
+    ]
+
+    dispatcher.send_large_table(header, rows)
+
+    return CommandStatusChoices.STATUS_SUCCEEDED
+
+
+@subcommand_of("nautobot")
+def get_jobs(dispatcher, kwargs: str = ""):
+    """Get all jobs from Nautobot that the requesting user have view permissions for.
+
+    Args:
+        kwargs (str): JSON-string array of header items to be exported. (Optional, default export is: name, id, enabled)
+    """
+    # Confirm kwargs is valid JSON
+    json_args = ["name", "id", "enabled"]
+    try:
+        if kwargs:
+            json_args = json.loads(kwargs)
+    except json.JSONDecodeError:
+        dispatcher.send_error(f"Invalid JSON-string, cannot decode: {kwargs}")
+        return (CommandStatusChoices.STATUS_FAILED, f"Invalid JSON-string, cannot decode: {kwargs}")
+
+    # Confirm `name` is always present in export
+    name_key = json_args.get("name") or json_args.get("Name")
+    if not name_key:
+        json_args.append("name")
+
+    jobs = Job.objects.restrict(dispatcher.user, "view").all()
+
+    # Check if all items in json_args are valid keys (assuming all keys of job object are valid)
+    valid_keys = [attr for attr in dir(Job) if not callable(getattr(Job, attr)) and not attr.startswith("_")]
+    for item in json_args:
+        if item.lower() not in valid_keys:
+            dispatcher.send_error(f"Invalid item provided: {item.lower()}")
+            return (CommandStatusChoices.STATUS_FAILED, f"Invalid item provided: {item.lower()}")
+
+    # TODO: Check json_args are all valid keys
+    header = [item.capitalize() for item in json_args]
+    rows = [(tuple(str(getattr(job, item, "")) for item in json_args)) for job in jobs]
+
+    dispatcher.send_large_table(header, rows)
+
+    return CommandStatusChoices.STATUS_SUCCEEDED
+
+
+@subcommand_of("nautobot")
+def init_job(dispatcher, *args, job_name: str = "", json_string_kwargs: str = ""):  # pylint: disable=too-many-locals
```

Comment:
Would this make more sense to be named `start_job` or maybe `launch_job`? And then as I put that into suggestion, I then looked at Nautobot itself. Probably `run_job` would make sense to line up with the Run Job button.

Reply:
`run_job` should be fine for the namespace. Initially I did not use `run_job` namespace since there was a similar import `nautobot.extras.jobs import run_job`, but this has been replaced by `enqueue_job` anyways in the PR